<a href="https://colab.research.google.com/github/gilesitorr/Estadistica_Avanzada/blob/main/Parcial_3/Tarea_2/EA_parcial3_tarea2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# __Estadística Avanzada__
# Parcial 3 - Tarea 2
## _Gil Estéfano Rodríguez Rivera_

Todas las bases de datos utilizadas así como el desarrollo se encuentra en _GitHub_ en https://github.com/gilesitorr/Estadistica_Avanzada/tree/main/Parcial_3/Tarea_2

__Nota:__ En este Notebook se usó el kernel de R, aunque la extensión sea ipynb.

In [ ]:
install.packages("agricolae")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependencies ‘sass’, ‘R.methodsS3’, ‘R.oo’, ‘R.utils’, ‘httpuv’, ‘xtable’, ‘fontawesome’, ‘sourcetools’, ‘promises’, ‘bslib’, ‘R.cache’, ‘e1071’, ‘shiny’, ‘miniUI’, ‘styler’, ‘classInt’, ‘labelled’, ‘combinat’, ‘questionr’, ‘klaR’, ‘AlgDesign’




In [ ]:
# Antes de comenzar, se importan las librerías a usar
library(dplyr)
library(agricolae)

## Biostatistics: A Foundation for Analysis in the Health Sciences
### Wayne W. Daniel & Chad L. Cross

### Ejemplo 13.3.3


__A sample of 15 patients suffering from asthma participated in an experiment to study the effect of a
new treatment on pulmonary function. Among the various measurements recorded were those of
forced expiratory volume (liters) in 1 second (FEV1) before and after application of the treatment.
The results were as follows:__

__On the basis of these data, can one conclude that the treatment is effective in increasing the FEV1
level? Let $\alpha=0.05$ and find the p value__

In [15]:
# Se leen los datos de cada sujeto
before <- c(1.69, 2.77, 1.00, 1.66, 3.00, 0.85, 1.42, 2.82,
            2.58, 1.84, 1.89, 1.91, 1.75, 2.46, 2.35) # Antes del tratamiento

after <- c(1.69, 2.22, 3.07, 3.35, 3.00, 2.74, 3.61, 5.14,
           2.44, 4.17, 2.42, 2.94, 3.04, 4.62, 4.42) # Luego del tratamiento

# Se calcula la diferencia
diferencia <- after-before

In [42]:
# Se cuentan los números positivos
no.cero <- length(diferencia[diferencia!=0])
negativos <- length(diferencia[diferencia<0])

# Se hace la prueba binomial para la cantidad de diferencias de signo positivo
binom.test(negativos, no.cero, alternative="l")


	Exact binomial test

data:  negativos and no.cero
number of successes = 2, number of trials = 13, p-value = 0.01123
alternative hypothesis: true probability of success is less than 0.5
95 percent confidence interval:
 0.0000000 0.4100986
sample estimates:
probability of success 
             0.1538462 


De acuerdo con la prueba del signo para 13 diferencias diferentes de 0 contra 2 valores menores a cero, se concluye que hay evidencia estadísticamente significativa ($\alpha=0.05$) para afirmar que el tratamiento si implica un aumento del nivel de FEV1 luego del tratamiento ($p=0.0112$). 

### Ejemplo 13.4.3


__In a study by Zuckerman and Heneghan (A-3), hemodynamic stresses were measured on subjects
undergoing laparoscopic cholecystectomy. An outcome variable of interest was the ventricular end
diastolic volume (LVEDV) measured in milliliters. A portion of the data appear in the following table.
Baseline refers to a measurement taken 5 minutes after induction of anesthesia, and the term “5
minutes” refers to a measurement taken 5 minutes after baseline.__

__May we conclude, on the basis of these data, that among subjects undergoing laparoscopic
cholecystectomy, the average LVEDV levels change? Let $\alpha=0.01$.__

In [ ]:
# Se define el url donde está guardado el archivo csv a usar
url = "https://github.com/gilesitorr/Estadistica_Avanzada/raw/main/Parcial_3/Tarea_2/EXR_C13_S04_03.csv"

# Se lee el archivo csv desde el url que se definió
df <- read.csv(url, header=TRUE)

In [ ]:
# La base de datos correcta
df

Subject Baseline X5_Minutes
1   1      51.7      49.3     
2   2      79.0      72.0     
3   3      78.7      87.3     
4   4      80.3      88.3     
5   5      72.0     103.3     
6   6      85.0      94.0     
7   7      69.7      94.7     
8   8      71.3      46.3     
9   9      55.7      71.7     
10 10      56.3      72.3

In [ ]:
# Se realiza una prueba de Wilcozon pareada porque cada sujeto
# de la muestra tuvo una medición base y otra luego de 5 minutos
wilcox.test(x=df$Baseline, y=df$X5_Minutes, paired=TRUE)

Warning message in wilcox.test.default(x = df$Baseline, y = df$X5_Minutes, paired = TRUE):
“cannot compute exact p-value with ties”



	Wilcoxon signed rank test with continuity correction

data:  df$Baseline and df$X5_Minutes
V = 11.5, p-value = 0.1137
alternative hypothesis: true location shift is not equal to 0


Con base a la prueba de Wilcoxon, a una significancia de $\alpha=0.01$, se infiere que no hay suficiente evidencia para afirmar que se produce un cambio estadísticamente significativo en los niveles promedio de LVEDV en los cinco minutos entre mediciones ($p=0.1137$).

### Ejemplo 13.5.1


__Fifteen patient records from each of two hospitals were reviewed and assigned a score designed to
measure level of care. The scores were as follows:__

Hospital A: 99, 85, 73, 98, 83, 88, 99, 80, 74, 91, 80, 94, 94, 98, 80

Hospital B: 78, 74, 69, 79, 57, 78, 79, 68, 59, 91, 89, 55, 60, 55, 79

__Would you conclude, at the .05 level of significance, that the two population medians are different?
Determine the p value.__

In [ ]:
# Se definen las medidas de ambas muestras
a <- c(99, 85, 73, 98, 83, 88, 99, 80, 74, 91, 80, 94, 94, 98, 80) # Hospital A
b <- c(78, 74, 69, 79, 57, 78, 79, 68, 59, 91, 89, 55, 60, 55, 79) # Hospital B

# Se separa por hospital y se genera un solo dataframe
df <- data.frame(x=c(a,b), y=c(rep("A", length(a)), rep("B", length(a))))


In [ ]:
# Se realiza la prueba de la mediana para ambos hospitales
Median.test(df$x,df$y)


The Median Test for df$x ~ df$y 

Chi Square = 16.13333   DF = 1   P.Value 5.903578e-05
Median = 79.5 

  Median  r Min Max  Q25 Q75
A     88 15  73  99 80.0  96
B     74 15  55  91 59.5  79

Post Hoc Analysis

Groups according to probability of treatment differences and alpha level.

Treatments with the same letter are not significantly different.

  df$x groups
A   88      a
B   74      b


De acuerdo a la prueba de la mediana, a una significancia de $0.05$, se concluye que hay suficiente evidencia para afirmar que la diferencia entre las medianas de las puntuaciones de ambos hospitales es estadíticamente significativa ($X^2=16.13$, $p\sim6\times10^{-5}$).

### Ejemplo 13.5.2


__The following serum albumin values were obtained from 17 normal and 13 hospitalized subjects:__

__Would you conclude at the .05 level of significance that the medians of the two populations sampled
are different? Determine the p value.__

In [ ]:
# Se definen las medidas de ambas muestras
normal <- c(2.4, 3.0, 3.4, 4.0, 3.5, 3.2, 4.5, 3.5, 3.1,
            3.5, 5.0, 3.6, 4.0, 3.8, 2.9, 4.2, 3.9)

hospital <- c(1.5, 3.1, 3.8, 1.5, 2.0, 1.3, 3.5,
              3.4, 1.5, 1.7, 1.8, 2.0, 2.0)

# Se separa por grupo y se genera un solo dataframe
df <- data.frame(x=c(normal,hospital),
                 y=c(rep("N", length(normal)), rep("H", length(hospital))))

In [ ]:
# Se realiza la prueba de la mediana para ambos grupos
Median.test(df$x,df$y)


The Median Test for df$x ~ df$y 

Chi Square = 6.651584   DF = 1   P.Value 0.009906773
Median = 3.3 

  Median  r Min Max Q25 Q75
H    2.0 13 1.3 3.8 1.5 3.1
N    3.5 17 2.4 5.0 3.2 4.0

Post Hoc Analysis

Groups according to probability of treatment differences and alpha level.

Treatments with the same letter are not significantly different.

  df$x groups
N  3.5      a
H  2.0      b


De acuerdo a la prueba de la mediana, a una significancia de $0.05$, se concluye que hay suficiente evidencia para afirmar que la diferencia entre las medianas de los niveles de albúmina de cada grupo de pacientes es estadíticamente significativa ($X^2=6.65$, $p=0.0099$).